src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -2875,6 +2875,1036 @@ void M4(bool b)
+
+class C
+{
+    public C? M0(out int x, out int y) { x = 42; y = 42; return this; }
+
+    public void M1(C c)
+    {
+        int x1, y1, x2, y2;
+        _ = c.M0(out x1, out y1)?.M0(out x2, out y2) != null
+            ? x1.ToString() + x2.ToString()
+            : y1.ToString() + y2.ToString(); // 1
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (13,31): error CS0165: Use of unassigned local variable 'y2'
+                //             : y1.ToString() + y2.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y2").WithArguments("y2").WithLocation(13, 31)
+                );
+        }
+
+        [Fact]
+        public void EqualsCondAccess_07()
+        {
+            var source = @"
+#nullable enable
+
+struct S
+{
+    public static bool operator ==(S left, S right) => false;
+    public static bool operator !=(S left, S right) => false;
+    public override bool Equals(object obj) => false;
+    public override int GetHashCode() => 0;
+
+    public S M0(out int x, out int y) { x = 42; y = 42; return this; }
+
+    public void M1(S? s)
+    {
+        int x, y;
+        _ = s?.M0(out x, out y) != null
+            ? x.ToString()
+            : y.ToString(); // 1
+    }
+
+    public void M2(S? s)
+    {
+        int x, y;
+        _ = s?.M0(out x, out y) == null
+            ? x.ToString() // 2
+            : y.ToString();
+    }
+
+    public void M3(S? s)
+    {
+        int x, y;
+        _ = null != s?.M0(out x, out y)
+            ? x.ToString()
+            : y.ToString(); // 3
+    }
+
+    public void M4(S? s)
+    {
+        int x, y;
+        _ = null == s?.M0(out x, out y)
+            ? x.ToString() // 4
+            : y.ToString();
+    }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (18,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(18, 15),
+                // (25,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(25, 15),
+                // (34,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 3
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(34, 15),
+                // (41,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 4
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(41, 15)
+                );
+        }
+
+        [Fact]
+        public void EqualsCondAccess_08()
+        {
+            var source = @"
+#nullable enable
+
+struct S
+{
+    public static bool operator ==(S left, S right) => false;
```

Comment:
Would be good to have an example with reference types instead of struct types.

Reply:
Have added test `EqualsCondAccess_18` for this scenario.